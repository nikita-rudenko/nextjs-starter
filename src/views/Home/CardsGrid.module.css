.grid {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;

  max-width: 800px;
  margin-top: 3rem;
}

@media (max-width: 600px) {
  .grid {
    width: 100%;
    flex-direction: column;
  }
}
