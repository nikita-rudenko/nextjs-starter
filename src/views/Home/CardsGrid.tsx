import Card from '@components/Card';

import styles from './CardsGrid.module.css';

type Card = {
  href: string;
  title: string;
  description: string;
};

const cards: Card[] = [
  {
    href: 'https://nextjs.org/docs',
    title: 'Documentation ðŸ“„',
    description: 'Find in-depth information about Next.js features and API.',
  },
  {
    href: 'https://nextjs.org/learn',
    title: 'Learn ðŸ’¡',
    description: 'Learn about Next.js in an interactive course with quizzes!',
  },
  {
    href: 'https://github.com/vercel/next.js/tree/master/examples',
    title: 'Examples ðŸ§©',
    description: 'Discover and deploy boilerplate example Next.js projects.',
  },
  {
    href: 'https://vercel.com/import?filter=next.js',
    title: 'Deploy âœ¨',
    description:
      'Instantly deploy your Next.js site to a public URL with Vercel.',
  },
];

function CardsGrid(): JSX.Element {
  return (
    <div className={styles.grid}>
      {cards.map(({ href, title, description }) => {
        return (
          <Card
            key={title}
            href={href}
            title={title}
            description={description}
          />
        );
      })}
    </div>
  );
}

export default CardsGrid;
