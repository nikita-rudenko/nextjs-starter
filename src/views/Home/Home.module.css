.description {
  text-align: center;
  line-height: 1.5;
  font-size: 1.5rem;
}
