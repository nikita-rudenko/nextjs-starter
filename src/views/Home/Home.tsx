import Title from '@components/Title';
import Layout from '@layout';

import CardsGrid from './CardsGrid';
import styles from './Home.module.css';

function Home(): JSX.Element {
  return (
    <Layout>
      <Layout.Content>
        <Title>
          Welcome to <a href="https://nextjs.org">Next.js!</a>
        </Title>

        <p className={styles.description}>Start building your Next project!</p>

        <button
          onClick={() => {
            window.alert('With typescript and Jest');
          }}
        >
          Test Button
        </button>

        <CardsGrid />
      </Layout.Content>

      <Layout.Footer />
    </Layout>
  );
}

export default Home;
