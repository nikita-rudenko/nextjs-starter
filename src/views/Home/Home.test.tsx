import { render, fireEvent } from '@tests/testUtils';

import Home from './Home';

describe('Home view', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Home />, {});
    window.alert = jest.fn();
    fireEvent.click(getByText('Test Button'));
    expect(window.alert).toHaveBeenCalledWith('With typescript and Jest');
  });
});
