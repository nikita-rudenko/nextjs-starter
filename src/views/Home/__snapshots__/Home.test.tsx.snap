// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <main
      class="content"
    >
      <h1
        class="title"
      >
        <span>
          Welcome to 
          <a
            href="https://nextjs.org"
          >
            Next.js!
          </a>
        </span>
      </h1>
      <p
        class="description"
      >
        Start building your Next project!
      </p>
      <button>
        Test Button
      </button>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h3>
            Documentation ðŸ“„
          </h3>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h3>
            Learn ðŸ’¡
          </h3>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3>
            Examples ðŸ§©
          </h3>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/import?filter=next.js"
        >
          <h3>
            Deploy âœ¨
          </h3>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://vercel.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
